### PR TITLE
Add XML Documentation to ResponseCachingMiddleware and ResponseCachingExtensions

### PR DESCRIPTION
**Pull request by Marusyk**
Added XML docs to middleware and extension method

Addresses #18831

Please review
Thank you in advance

### CODE REVIEW

File: src/Middleware/ResponseCaching/src/ResponseCachingExtensions.cs [hunk 1]
```diff
@@ -6,8 +6,15 @@
 
 namespace Microsoft.AspNetCore.Builder
 {
+    /// <summary>
+    /// Extension methods for the ResponseCachingMiddleware middleware.
+    /// </summary>
     public static class ResponseCachingExtensions
     {
+        /// <summary>
+        /// Adds the <see cref="ResponseCachingMiddleware"/> for caching HTTP Responses.
```
Comment (anurse) [inline src/Middleware/ResponseCaching/src/ResponseCachingExtensions.cs]:
```suggestion
        /// Adds the <see cref="ResponseCachingMiddleware"/> for caching HTTP responses.
```

nit: Responses shouldn't be capitalized

File: src/Middleware/ResponseCaching/src/ResponseCachingExtensions.cs [hunk 2]
```diff
@@ -6,8 +6,15 @@
 
 namespace Microsoft.AspNetCore.Builder
 {
+    /// <summary>
+    /// Extension methods for the ResponseCachingMiddleware middleware.
```
Comment (anurse) [inline src/Middleware/ResponseCaching/src/ResponseCachingExtensions.cs]:
```suggestion
    /// Extension methods for adding the <see cref="ResponseCachingMiddleware"/> to an application.
```

Consistency with [other places](https://github.com/dotnet/aspnetcore/blob/87629bbad906e9507026692904b6bcb5021cdd33/src/Middleware/Session/src/SessionMiddlewareExtensions.cs#L11).

File: src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs [hunk 3]
```diff
@@ -29,6 +32,13 @@ public class ResponseCachingMiddleware
         private readonly IResponseCache _cache;
         private readonly IResponseCachingKeyProvider _keyProvider;
 
+        /// <summary>
+        /// Creates a new <see cref="ResponseCachingMiddleware"/>.
+        /// </summary>
+        /// <param name="next">The <see cref="RequestDelegate"/> representing the next middleware in the pipeline.</param>
+        /// <param name="options">The <see cref="ResponseCachingOptions"/> representing the options for this middleware.
```
Comment (anurse) [inline src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs]:
```suggestion
        /// <param name="options">The options for this middleware.
```

I think it's OK to not use a `<see>` tag here. I know the others do, but I'm not 100% on the appropriate syntax to reference `IOptions<ResponseCachingOptions>` (given the generics) and the docs will have the full signature and will cross-ref all the parameters types automatically anyway.

File: src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs [hunk 4]
```diff
@@ -29,6 +32,13 @@ public class ResponseCachingMiddleware
...
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> used for logging.</param>
+        /// <param name="poolProvider">The <see cref="ObjectPoolProvider"/> used for creating <see cref="ObjectPool"/>.</param>
```
Comment (anurse) [inline src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs]:
```suggestion
        /// <param name="poolProvider">The <see cref="ObjectPoolProvider"/> used for creating <see cref="ObjectPool"/> instances.</param>
```

**Review by analogrelay — COMMENTED** [review]
Some small edits but looks like a great contribution!

**Review by analogrelay — APPROVED** [review]

**Pull request merged**